src/xlsx/xlsx_writer.ts

```diff
@@ -142,19 +147,22 @@ function createWorksheets(data: ExcelWorkbookData, construct: XLSXStructure): XL
 
     const images = sheet.images;
     for (const image of images) {
+      const contentType = image.data.contentType ?? (await getImageContentType(image.data.path));
```

Comment:
I would probably pre-fetch all the `img.contentType` in `model`, so we don't have to worry about async functions and whatnot in the `xlsx_writer`. Just my opinon tho.